query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverterTest.java

```diff
@@ -207,37 +207,26 @@ void convertsConcatFunction() {
   @Test
   void convertAvgRateFunction() {
     Expression column1 = createColumnExpression("foo").build();
-    Expression column2 = createStringLiteralValueExpression("PT2S");
+    Expression column2 = createStringLiteralValueExpression("PT5S");
 
     when(this.mockArgumentConverter.apply(column1)).thenReturn("foo");
     when(this.mockingExecutionContext.getTimeSeriesPeriod())
         .thenReturn(Optional.of(Duration.ofSeconds(10)));
 
     assertEquals(
-        "SUM(DIV(foo, 5.0))",
+        "SUM(DIV(foo, 2.0))",
         new PinotFunctionConverter()
             .convert(
                 mockingExecutionContext,
                 buildFunction(QUERY_FUNCTION_AVG_RATE, column1.toBuilder(), column2.toBuilder()),
                 this.mockArgumentConverter));
 
     when(this.mockingExecutionContext.getTimeSeriesPeriod()).thenReturn(Optional.empty());
-    when(this.mockingExecutionContext.getTimeRangeDuration()).thenReturn(Duration.ofSeconds(10));
+    when(this.mockingExecutionContext.getTimeRangeDuration())
+        .thenReturn(Optional.of(Duration.ofSeconds(1)));
 
     assertEquals(
-        "SUM(DIV(foo, 5.0))",
-        new PinotFunctionConverter()
-            .convert(
-                mockingExecutionContext,
-                buildFunction(QUERY_FUNCTION_AVG_RATE, column1.toBuilder(), column2.toBuilder()),
-                this.mockArgumentConverter));
-
-    when(this.mockingExecutionContext.getTimeSeriesPeriod())
-        .thenReturn(Optional.of(Duration.ofSeconds(20)));
-    when(this.mockingExecutionContext.getTimeRangeDuration()).thenReturn(Duration.ofSeconds(10));
-
-    assertEquals(
-        "SUM(DIV(foo, 10.0))",
+        "SUM(DIV(foo, 0.2))",
```

Comment:
I'm not saying we need to put in validation for it, but we don't want rate intervals that are larger than the time range - it doesn't make semantic sense (e.g. what's my rate per hour over the last minute) and would produce incorrect results. We certainly don't need to test for it.